addons/web/static/src/js/views/list/list_confirm_dialog.js

```diff
@@ -68,25 +68,53 @@ const ListConfirmDialog = Dialog.extend(WidgetAdapterMixin, {
                 options: widgetOptions,
             });
         }
+
+        this.relatedFieldWidget = undefined;
+        if (this.fieldWidget.attrs.widget === 'daterange' && changes.relatedFieldName) {
+            const fieldName = changes.relatedFieldName;
+            const fieldLabel = record.fields[fieldName].string;
+            this.$content.find('table.o_modal_changes tbody').append(qweb.render('ListView.confirmModal.relatedField', { fieldLabel }));
+            const RelatedWidget = record.fieldsInfo.list[fieldName].Widget;
+
+            if (this.isLegacyWidget) { // the related widget is the same (= DateRange widget) then we don't need to recheck it
+                this.relatedFieldWidget = new RelatedWidget(this, fieldName, record, widgetOptions);
+            } else {
+                this.relatedFieldWidget = new FieldWrapper(this, Widget, {
+                    fieldName,
+                    record,
+                    options: widgetOptions,
+                });
+            }
```

Comment:
Couldn't we maybe extract that logic (and the one from [here](https://github.com/odoo/odoo/pull/69892/commits/f40b5170ffccdd2a2ebc3eb66e5351ef3b2aab81#diff-45471d4fec30f084ed5518b5a231b7e9b7ad0b39c82cae723ff64b1e883ba534R55) and from [here](https://github.com/odoo/odoo/pull/69892/commits/f40b5170ffccdd2a2ebc3eb66e5351ef3b2aab81#diff-45471d4fec30f084ed5518b5a231b7e9b7ad0b39c82cae723ff64b1e883ba534L62-L70)) into a function and reuse it both locations by providing according params ?

Reply:
Sorry, since I have forced push in this branch I don't have the correct content that you wanted to show me 😩 
But I suppose it is as I propose here : https://github.com/odoo/odoo/pull/69892/files/f40b5170ffccdd2a2ebc3eb66e5351ef3b2aab81#r626542181